Formula/gobject-introspection-utils.rb

```diff
@@ -0,0 +1,101 @@
+  homepage "https://gi.readthedocs.io/en/latest/"
+  url "https://download.gnome.org/sources/gobject-introspection/1.72/gobject-introspection-1.72.0.tar.xz"
+  sha256 "02fe8e590861d88f83060dd39cda5ccaa60b2da1d21d0f95499301b186beaabc"
+  license all_of: ["GPL-2.0-or-later", "LGPL-2.0-or-later", "MIT"]
+
+  depends_on "bison" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "cairo"
+  depends_on "glib"
+  depends_on "gobject-introspection"
+  depends_on "libffi"
+  depends_on "pkg-config"
+  depends_on "python@3.10"
+
+  uses_from_macos "flex" => :build
+
+  def install
+    ENV["GI_SCANNER_DISABLE_CACHE"] = "true"
+    inreplace "giscanner/transformer.py", "/usr/share", "#{HOMEBREW_PREFIX}/share"
+    inreplace "meson.build",
+      "config.set_quoted('GOBJECT_INTROSPECTION_LIBDIR', join_paths(get_option('prefix'), get_option('libdir')))",
+      "config.set_quoted('GOBJECT_INTROSPECTION_LIBDIR', '#{HOMEBREW_PREFIX}/lib')"
+
+    # Point the headers and libraries to `gobject-introspection`.
+    # The headers and libraries will be removed later because they are provided by `gobject-introspection`.
+    gobject_introspection = Formula["gobject-introspection"]
+    args = std_meson_args.delete_if do |arg|
+      arg.start_with?("--includedir=", "--libdir=")
+    end
+    args += %W[
+      --includedir=#{gobject_introspection.opt_include}
+      --libdir=#{gobject_introspection.opt_lib}
+    ]
+
+    mkdir "build" do
+      system "meson", *args,
+        "-Dpython=#{Formula["python@3.10"].opt_bin}/python3",
+        ".."
+      system "ninja", "-v"
+
+      # Skip files already in gobject-introspection.
+      Formula["meson"].opt_libexec.cd do
+        system "bin/python3", "-c", pyscript
+      end
+      system "ninja", "install", "-v"
+    end
+
+    # Delete non python files because they are provided by `gobject-introspection`
+    python_extension_regex = /\.(py(?:[diwx])?|px[di]|cpython-(?:[23]\d{1,2})[-\w]*\.(so|dylib))$/i
+    python_shebang_regex = %r{^#! ?/usr/bin/(?:env )?python(?:[23](?:\.\d{1,2})?)?( |$)}
+    shebang_max_length = 28 # the length of "#! /usr/bin/env pythonx.yyy "
+    prefix.find do |f|
+      next unless f.file?
+      next if python_extension_regex.match?(f.basename) || python_shebang_regex.match?(f.read(shebang_max_length))
+
+      f.unlink
+    end
+
+    # Delete empty directories
+    # Note: We need to traversal the directories in reverse order (i.e. deepest first).
+    #       Also, we should put checking emptiness and deletion in a single loop.
+    #       `dirs.select(&:empty?).map(&:rmdir)` will not work because it will not delete
+    #       directories that only contain empty directories.
+    prefix.find.select(&:directory?).reverse_each { |d| d.rmdir if d.empty? }
+
+    # Install giscanner Python library.
+    (lib/"gobject-introspection/giscanner").install (buildpath/"build/giscanner").glob(["*.py", "*.so", "*.a"])
```

Comment:
In that case I think I have an idea of how to tweak the `pyscript` method to make sure it doesn't remove `giscanner` from the install targets in `gobject-introspection-utils`.  I'll try to push a commit for this later today.

Reply:
I was able to fix this so that we don't have to manually install any DSOs.  I didn't have to tweak anything with `pyscript`, just the arguments to `meson`.